### PR TITLE
Documented \eZ\Publish\API\Repository\SettingService exceptions

### PR DESCRIPTION
**Pull request by adamwojs**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | -
| **Type**                                   | improvement
| **Target eZ Platform version** | `v3.3`
| **BC breaks**                          | no
| **Doc needed**                       | no

#### Checklist:
- [ ] ~Provided PR description.~
- [ ] ~Tested the solution manually.~
- [ ] ~Provided automated test coverage.~
- [X] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [X] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [ ] ~Asked for a review (ping `@ezsystems/php-dev-team`).~

### CODE REVIEW

File: eZ/Publish/API/Repository/SettingService.php [hunk 1]
```diff
@@ -14,12 +14,27 @@
...
 {
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If setting with the given group and identifier could not be found
+     */
     public function loadSetting(string $group, string $identifier): Setting;
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user user is not allowed to update a setting
+     */
```
Comment (Steveb-p) [inline eZ/Publish/API/Repository/SettingService.php:25]: `updateSetting` method should also be declared as potentially throwing `NotFoundException` (when failing to update a setting that is for example already deleted), despite not being ever thrown in the actual implementation (yet?).

In the same manner, as discussed, in my opinion `InvalidArgumentException` should remain, despite never being thrown from implementation atm.

File: eZ/Publish/API/Repository/SettingService.php [hunk 2]
```diff
@@ -14,12 +14,27 @@
...
     public function updateSetting(Setting $setting, SettingUpdateStruct $settingUpdateStruct): Setting;
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If setting with the given group and identifier already exists
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user user is not allowed to create a setting
```
Comment (DominikaK) [inline eZ/Publish/API/Repository/SettingService.php:30]:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user is not allowed to create a setting
```

File: eZ/Publish/API/Repository/SettingService.php [hunk 3]
```diff
@@ -14,12 +14,27 @@
...
+     */
     public function createSetting(SettingCreateStruct $settingCreateStruct): Setting;
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If setting has already been removed
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user user is not allowed to remove a setting
```
Comment (DominikaK) [inline eZ/Publish/API/Repository/SettingService.php:36]:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user is not allowed to remove a setting
```

File: eZ/Publish/API/Repository/SettingService.php [hunk 4]
```diff
@@ -14,12 +14,27 @@
 
 interface SettingService
 {
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If setting with the given group and identifier could not be found
```
Comment (DominikaK) [inline eZ/Publish/API/Repository/SettingService.php:18]:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If a setting with the given group and identifier could not be found
```

File: eZ/Publish/API/Repository/SettingService.php [hunk 5]
```diff
@@ -14,12 +14,27 @@
...
     public function loadSetting(string $group, string $identifier): Setting;
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user user is not allowed to update a setting
+     */
     public function updateSetting(Setting $setting, SettingUpdateStruct $settingUpdateStruct): Setting;
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If setting with the given group and identifier already exists
```
Comment (DominikaK) [inline eZ/Publish/API/Repository/SettingService.php:29]:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If a setting with the given group and identifier already exists
```

File: eZ/Publish/API/Repository/SettingService.php [hunk 6]
```diff
@@ -14,12 +14,27 @@
...
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user user is not allowed to create a setting
+     */
     public function createSetting(SettingCreateStruct $settingCreateStruct): Setting;
 
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If setting has already been removed
```
Comment (DominikaK) [inline eZ/Publish/API/Repository/SettingService.php:35]:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the setting has already been removed
```

**Comment by adamwojs** [comment]
PR updated according to code review suggestions.

**Review by Steveb-p — APPROVED** [review]

**Pull request merged**